pkg/disttask/framework/storage/task_table.go

```diff
@@ -607,10 +607,14 @@ func (mgr *TaskManager) SwitchTaskStep(
 ) error {
 	return mgr.WithNewTxn(ctx, func(se sessionctx.Context) error {
 		vars := se.GetSessionVars()
-		if vars.MemQuotaQuery < variable.DefTiDBMemQuotaQuery {
+		// in most cases the def memory quota is enough, but when importing 200TiB
+		// data using global sort, when transiting to write&ingest step, the txn
+		// might be cancelled, so we double it.
+		targetQuota := 2 * variable.DefTiDBMemQuotaQuery
```

Comment:
Is there an unlimited option?

Reply:
https://docs.pingcap.com/tidb/stable/system-variables#tidb_mem_quota_query , 2G should be enough for datasize below 400T